src/main/java/ruslan/shastkiv/bookstore/controller/BookController.java

```diff
@@ -42,17 +44,34 @@ public List<BookDto> getAll(Pageable pageable) {
             description = "Retrieve a specific book by its ID. "
                     + "Returns 404 if the book is not found."
     )
+    @PreAuthorize("hasRole('USER')")
     @GetMapping("/{id}")
     public BookDto getBokById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
+    @Operation(
+            summary = "Search for books",
+            description = "Search for books by author, title, or price range. "
+                    + "If only one price is provided, it will be treated as the minimum price, "
+                    + "and results will be returned starting from that price. "
+                    + "If two prices are provided, they will be treated as "
+                    + "the minimum and maximum price range."
+    )
+    @PreAuthorize("hasRole('USER')")
+    @GetMapping("/search")
+    public List<BookDto> search(BookSearchParametersDto searchParametersDto, Pageable pageable) {
```

Comment:
check other Pageable
```suggestion
    public List<BookDto> search(BookSearchParametersDto searchParametersDto, @ParameterObject @PageableDefault Pageable pageable) {
```